### DISCUSSION TITLE
Upload with HFS3

### DISCUSSION THREAD

**Discussion by quite68** [rejetto/hfs#143] [Q&A] [answered]
How do you do upload with HFS3? Is there some documentation somewhere?

I'm just mucking around with a local server. The following would work with HFS2

curl -v -F key1=value1 -F upload=@x.x localhost:8011

But with HFS3 I get

* Rebuilt URL to: localhost:8011/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8011 (#0)
> POST / HTTP/1.1
> Host: localhost:8011
> User-Agent: curl/7.58.0
> Accept: */*
> Content-Length: 326
> Content-Type: multipart/form-data; boundary=------------------------f583a94f1df5c8d8
> 
< HTTP/1.1 401 Unauthorized
< Vary: Accept-Encoding
< Content-Type: application/json; charset=utf-8
< Content-Length: 2
< Date: Fri, 10 Mar 2023 16:31:09 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
* HTTP error before end of send, stop sending
< 
* Closing connection 0

**Comment by rejetto** [comment, marked as answer]
this is the easiest way `curl -T file url/`
url includes the folder
of course you need to give upload permission to that folder

**Reply by quite68** [reply]
Thanks that looks more promising

curl -T x.x localhost:8011/

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 28    42  100    12    0     0    800      0 --:--:-- --:--:-- --:--:--   800

However I can't find the file.
How do I give a folder upload permissions?

**Reply by rejetto** [reply]
in "shared files", select the folder, and "who can upload"
for your tests, start without the need of an account, let upload to "anyone".

beware:  only folders with a "source" can get uploads.

You can verify that you have upload permission by browsing the right folder and by seeing the "upload" button appear.
<img width="192" alt="image" src="https://user-images.githubusercontent.com/1367199/224389130-273c8cfb-582c-4eba-8ad8-9b04c9923de9.png">
